--- 2010_0709_RT2870_Linux_STA_v2.4.0.1/common/rtusb_dev_id.c	2010-07-09 04:13:47.000000000 +0200
+++ common/rtusb_dev_id.c	2010-09-26 14:50:18.000000000 +0200
@@ -90,7 +90,7 @@
 	{USB_DEVICE(0x129B,0x1828)}, /* Siemens */
 	{USB_DEVICE(0x157E,0x300E)},	/* U-Media */
 	{USB_DEVICE(0x050d,0x805c)},	
-	{USB_DEVICE(0x050d,0x815c)},
+       {USB_DEVICE(0x1761,0x0b05)},
 	{USB_DEVICE(0x1482,0x3C09)}, /* Abocom*/
 	{USB_DEVICE(0x14B2,0x3C09)}, /* Alpha */
 	{USB_DEVICE(0x04E8,0x2018)}, /* samsung linkstick2 */
