--- 2010_1215_RT3572_Linux_STA_v2.5.0.0.DPO/common/rtusb_dev_id.c 2010-07-09 04:09:41.000000000 +0200 
+++ 2010_1215_RT3572_Linux_STA_v2.5.0.0.DPO/common/rtusb_dev_id.c 2010-10-27 21:53:02.931136482 +0200 
@@ -122,6 +122,7 @@
 	{USB_DEVICE(0x5A57,0x0284)}, /* Zinwell 3572 */
 	{USB_DEVICE(0x167B,0x4001)}, /* 3572 */
 	{USB_DEVICE(0x0930,0x0A07)}, /* TOSHIBA */
+	{USB_DEVICE(0x1737,0x0079)}, /* Linksys WUSB600N v2 */
 #endif /* RT35xx */
 	{ }/* Terminating entry */
 };

