diff -Naur git-org/src/vm/vm.c git/src/vm/vm.c
--- git-org/src/vm/vm.c	2010-06-07 11:28:53.000000000 +0200
+++ git/src/vm/vm.c	2010-12-13 10:08:11.765126889 +0100
@@ -1662,6 +1662,7 @@
 /* Figure out the correct pgN from the cell and update (vm->state). */
 static int set_PGN(vm_t *vm) {
   int new_pgN = 0;
+  int dummy, part;
 
   while(new_pgN < (vm->state).pgc->nr_of_programs
 	&& (vm->state).cellN >= (vm->state).pgc->program_map[new_pgN])
@@ -1678,14 +1679,8 @@
     if((vm->state).TTN_REG > vm->vmgi->tt_srpt->nr_of_srpts)
       return 0; /* ?? */
     pb_ty = &vm->vmgi->tt_srpt->title[(vm->state).TTN_REG - 1].pb_ty;
-    if(pb_ty->multi_or_random_pgc_title == /* One_Sequential_PGC_Title */ 0) {
-      int dummy, part;
-      vm_get_current_title_part(vm, &dummy, &part);
-      (vm->state).PTTN_REG = part;
-    } else {
-      /* FIXME: Handle RANDOM or SHUFFLE titles. */
-      fprintf(MSG_OUT, "libdvdnav: RANDOM or SHUFFLE titles are NOT handled yet.\n");
-    }
+    vm_get_current_title_part(vm, &dummy, &part);
+    (vm->state).PTTN_REG = part;
   }
   return 1;
 }
