diff -Naur perl-5.8.8.orig/Configure perl-5.8.8/Configure
--- perl-5.8.8.orig/Configure	2006-01-08 15:51:03.000000000 +0100
+++ perl-5.8.8/Configure	2011-06-30 22:07:51.016047904 +0200
@@ -1263,6 +1263,7 @@
 glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
 test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
 test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
+test -f /usr/lib/*-linux-gnu/libc.so && glibpth="/usr/lib/*-linux-gnu $glibpth"
 
 : Private path used by Configure to find libraries.  Its value
 : is prepended to libpth. This variable takes care of special
